lib/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVM.cpp

```diff
@@ -1611,6 +1611,118 @@ struct AbsFOpConversion
+    // NOTE: This pattern matching is not general enough, but it is sufficient.
+    // We detect only two cases here:
+    // 1. where the "-limit" is computed as 0 - limit:
+    //   %cst = arith.constant dense<0.000000e+00>
+    //   %8 = tt.load %7, %2
+    //   %11 = arith.subf %cst, %8
+    //   %12 = tt.clamp %5, %11, %8
+    // 2. where "-limit" and "limit" are constants.
+    //   %cst_6 = arith.constant dense<-6.0000e+00>
+    //   %cst_7 = arith.constant dense<6.0000e+00>
+    //   %160 = tt.clamp %158, %cst_6, %cst_7
+    bool clipPatternFound = false;
+
+    auto getSplatInitializer = [](Value v) -> std::optional<double> {
+      if (auto constOp = llvm::dyn_cast<arith::ConstantOp>(v.getDefiningOp())) {
+        if (auto attr = constOp.getValueAttr()
+                            .dyn_cast<mlir::DenseIntOrFPElementsAttr>()) {
+          if (attr.isSplat()) {
+            return attr.getSplatValue<APFloat>().convertToDouble();
+          }
+        }
+      }
+      return std::nullopt;
+    };
+
+    if (xorsignAbsAvailable) {
+      if (auto subOp =
+              llvm::dyn_cast<arith::SubFOp>(op.getOperand(1).getDefiningOp())) {
+        if (subOp.getOperand(1) == op.getOperand(2)) {
+          auto initializer = getSplatInitializer(subOp.getOperand(0));
+          if (initializer.has_value() && initializer.value() == 0.0) {
+            clipPatternFound = true;
+          }
+        }
+      } else if (llvm::isa<arith::ConstantOp>(
+                     op.getOperand(1).getDefiningOp()) &&
+                 llvm::isa<arith::ConstantOp>(
+                     op.getOperand(2).getDefiningOp())) {
+        auto initializer1 = getSplatInitializer(op.getOperand(1));
+        auto initializer2 = getSplatInitializer(op.getOperand(2));
+        if (initializer1.has_value() && initializer2.has_value() &&
+            initializer1.value() == -initializer2.value()) {
+          clipPatternFound = true;
+        }
+      }
+    }
+
+    assert(elemTy.isF32() || elemTy.isF16());
+
+    if (clipPatternFound) {
+      // min.xorsign.abs
+      PTXBuilder ptxBuilder;
+      bool propNan = (op.getPropagateNan() == mlir::triton::PropagateNan::ALL);
+      auto &minXorsign = ptxBuilder.create<PTXInstr>("min")
+                             ->o("NaN", propNan)
+                             .o("xorsign")
+                             .o("abs");
+      const char *outType = nullptr;
+      const char *inType = nullptr;
+      if (elemTy.isF32()) {
+        minXorsign.o("f32");
+        outType = "=f";
+        inType = "f";
+      } else if (elemTy.isF16()) {
+        minXorsign.o("f16");
+        outType = "=h";
+        inType = "h";
+      }
+      auto output = ptxBuilder.newOperand(outType);
+      auto inputA = ptxBuilder.newOperand(operands[0][0], inType);
+      auto inputB = ptxBuilder.newOperand(operands[0][2], inType);
+      minXorsign(output, inputA, inputB);
+
+      return {ptxBuilder.launch(rewriter, loc, elemTy, false)};
+    } else {
```

Comment:
nit: skip else after early return, that makes it a bit easier to read and reduce indentation